lib/elixir/lib/code.ex

```diff
@@ -1197,6 +1197,105 @@ defmodule Code do
     :elixir.string_to_quoted!(to_charlist(string), line, column, file, opts)
   end
 
+  @doc """
+  Converts the given string to its quoted form and a list of comments.
+
+  This function is useful when performing textual changes to the source code,
+  while preserving information like comments and literals position.
+
+  Returns `{:ok, quoted_form, comments}` if it succeeds,
+  `{:error, {line, error, token}}` otherwise.
+
+  Comments are maps containing information about the line they were found, its
+  contents, and how many end of lines were found between the comment and the
+  closest tokens:
+      iex> Code.string_to_quoted_with_comments("\""
+      ...> :foo
+      ...>
+      ...> # Hello, world!
+      ...>
+      ...>
+      ...> # Some more comments!
+      ...> "\"")
+      {:ok, :foo, [
+        %{line: 3, previous_eol: 2, next_eol: 3, text: "\# Hello, world!"},
```

Comment:
It took me a while to understand what `previous_eol`/`next_eol` meant. I think we need to mention the keys by name in the docs and I would suggest to rename them to something else. Some alternatives:

* `previous_eol_count`/`next_eol_count`
* `previous_eols`/`next_eols`

Thoughts?

Reply:
Maybe they could be renamed to `previous_eols`/`next_eols` and the docs could explain what each field represents:
> * `:line` - The line number the source code
> * `:text` - The full text of the comment, incluing the leading `#`
> * `:previous_eols` - How many end of lines there are between the comment and the previous node
> * `:next_eols` - How many end of lines there are between the comment and the next node

Another alternative could be `leading_eols`/`trailing_eols`?

Reply:
The docs are definitely welcome! On leading vs trailing eols, it also works. @whatyouhide any preferences?

Reply:
I’m fine with both but I personally read better with `previous_eol_count`, but I don't care too much :) But the docs are definitely welcome, agreed.

Reply:
Done :)